src/tools/plan-generator.ts

```diff
@@ -91,15 +94,30 @@ export class PlanGenerator {
 
   /** Generates a Kotlin `Plan.HandleConnection` from a HandleConnection. */
   async createHandleConnection(connection: HandleConnection): Promise<string> {
+    const schemaRegistry = [];
+    if (connection.type.hasVariable) {
+      schemaRegistry.push(await this.createSchemaForVariableHandleConnection(connection));
+    }
+
     const storageKey = await this.createStorageKey(connection.handle);
     const mode = this.createHandleMode(connection.direction, connection.type);
-    const type = generateConnectionType(connection);
+    const type = generateConnectionType(connection, schemaRegistry);
     const annotations = PlanGenerator.createAnnotations(connection.handle.annotations);
 
     return ktUtils.applyFun('HandleConnection', [storageKey, mode, type, annotations],
         {startIndent: 24});
   }
 
+  // Generate schemas for connections that have a type variable.
+  private async createSchemaForVariableHandleConnection(connection: HandleConnection): Promise<{schema: Schema, generation: string}> {
+      connection.type.maybeEnsureResolved();
+      const schema = connection.type.resolvedType().getEntitySchema();
+      const genNode = new SchemaNode(schema, connection.particle.spec, []);
```

Comment:
You are forced to go this roundabout way via creating a SchemaNode (which is supposed to be the output of Schema2Graph) in order to generate the schema. This should not be needed ideally, but that would be another refactoring to clean this up. The SchemaDescriptor is used both in cases where schemas are considered in abstraction from the Particle (like here) and where they relation to the particle is important (where having the SchemaNode is relevant).

I am against shortcuts in general, but given this is blocking Satakshi from progressing and given that's 2 lines of code that ideally should not be here, I think we should go forward as is and refactor in the future when we get bitten by this again, which may happen soon as recipe2plan may need to generate additional schema types for handles.